test/napi/src/js/objects.rs

```diff
@@ -167,6 +167,105 @@ pub fn return_external_array_buffer(mut cx: FunctionContext) -> JsResult<JsArray
     Ok(buf)
 }
 
+pub fn return_int8array_from_arraybuffer(mut cx: FunctionContext) -> JsResult<JsInt8Array> {
+    let buf = cx.argument::<JsArrayBuffer>(0)?;
+    JsInt8Array::from_buffer(&mut cx, buf)
+}
+
+pub fn return_int16array_from_arraybuffer(mut cx: FunctionContext) -> JsResult<JsInt16Array> {
+    let buf = cx.argument::<JsArrayBuffer>(0)?;
+    JsInt16Array::from_buffer(&mut cx, buf)
+}
+
+pub fn return_uint32array_from_arraybuffer(mut cx: FunctionContext) -> JsResult<JsUint32Array> {
+    let buf = cx.argument::<JsArrayBuffer>(0)?;
+    JsUint32Array::from_buffer(&mut cx, buf)
+}
+
+pub fn return_float64array_from_arraybuffer(mut cx: FunctionContext) -> JsResult<JsFloat64Array> {
+    let buf = cx.argument::<JsArrayBuffer>(0)?;
+    JsFloat64Array::from_buffer(&mut cx, buf)
+}
+
+pub fn return_biguint64array_from_arraybuffer(
+    mut cx: FunctionContext,
+) -> JsResult<JsBigUint64Array> {
+    let buf = cx.argument::<JsArrayBuffer>(0)?;
+    JsBigUint64Array::from_buffer(&mut cx, buf)
+}
+
+pub fn return_new_int32array(mut cx: FunctionContext) -> JsResult<JsInt32Array> {
+    let len = cx.argument::<JsNumber>(0)?.value(&mut cx) as usize;
+    JsInt32Array::new(&mut cx, len)
+}
+
+pub fn return_uint32array_from_arraybuffer_region(
+    mut cx: FunctionContext,
+) -> JsResult<JsUint32Array> {
+    let buf = cx.argument::<JsArrayBuffer>(0)?;
+    let byte_offset = cx.argument::<JsNumber>(1)?;
+    let byte_offset = byte_offset.value(&mut cx);
+    let len = cx.argument::<JsNumber>(2)?;
+    let len = len.value(&mut cx);
+    JsUint32Array::from_buffer_region(&mut cx, buf, byte_offset as usize, len as usize)
+}
+
+fn typed_array_info<'cx, C, T: Binary>(
+    cx: &mut C,
+    a: Handle<'cx, JsTypedArray<T>>,
+) -> JsResult<'cx, JsObject>
+where
+    C: Context<'cx>,
+{
+    let byte_offset = a.byte_offset(cx);
+    let byte_offset = cx.number(byte_offset as u32);
+
+    let len = a.len(cx);
+    let len = cx.number(len as u32);
+
+    let byte_length = a.byte_length(cx);
+    let byte_length = cx.number(byte_length as u32);
+
+    let buffer = a.buffer(cx);
+
+    let obj = cx.empty_object();
+
+    obj.set(cx, "byteOffset", byte_offset)?;
+    obj.set(cx, "length", len)?;
+    obj.set(cx, "byteLength", byte_length)?;
+    obj.set(cx, "buffer", buffer)?;
+
+    Ok(obj)
+}
+
+pub fn get_typed_array_info(mut cx: FunctionContext) -> JsResult<JsObject> {
```

Comment:
This implementation is unfortunate, but it's I think a pretty hard-to-avoid consequence of the dynamic/static divide between JavaScript and Rust.

One option would be to have some kind of a more dynamic super type of `JsTypedArray<T>` like `JsAnyTypedArray` with some kind of dynamic API for introspecting on the element type. But if we created an enum for the element type, it wouldn't be very future-proof in case JavaScript adds new array types (like 128 bit integers or 16 bit floats, for example). That would add new variants to the enum, which wouldn't be backwards-compatible.

Maybe it could be made to work with some kind of even more clever use of traits or something, not sure.

But this is a pretty peculiar function needed for the sake of testing Neon, and my gut tells me it'll be fairly rare that people need to write (a) generic code that operates on all `JsTypedArray<T>` and (b) expose that code directly to JavaScript as a JS API that provides generic typed array functionality.